src/ngrams/programming.rs

```diff
@@ -0,0 +1,84 @@
+use crate::ngrams::NgramData;
+
+use itertools::Itertools;
+
+pub struct ProgrammingData();
+
+// Source: https://github.com/e3b0c442/keywords/blob/main/chart.yaml
+/*
+    Python processing of that document after removing duplicate languages. Kept the newest versions.
+
+    from collections import Counter
+    import yaml
+
+    data = open(/path/to/yaml).read()
+    parsed = yaml.unsafe_load(data)
+    keywords = Counter()
+    for item in parsed:
+        keywords.update([str(word).lower() for word in item["keywords"]])
+    for word in sorted(word for word, _ in keywords.most_common(100)):
+        print(word)
+*/
+#[rustfmt::skip]
+const PROGRAMMING_KEYWORDS: [&str; 100] = [
+    "abstract", "and", "as", "assert", "async", "auto", "await", "bool", "break", "byte",
+    "case", "catch", "char", "class", "const", "continue", "def", "default", "do", "double",
+    "else", "elseif", "end", "enum", "explicit", "extends", "extern", "false", "final", "finally",
+    "float", "for", "from", "function", "get", "global", "goto", "if", "implements", "import",
+    "in", "inline", "instanceof", "int", "interface", "is", "let", "long", "match", "module",
+    "namespace", "new", "nil", "not", "null", "object", "open", "operator", "or", "out",
+    "override", "package", "private", "protected", "public", "register", "repeat", "return", "sealed", "select",
+    "self", "set", "short", "signed", "sizeof", "static", "struct", "super", "switch", "then",
+    "this", "throw", "true", "try", "type", "typedef", "typeof", "union", "unsigned", "until",
+    "using", "var", "void", "volatile", "when", "where", "while", "with", "xor", "yield",
+];
+
```

Comment:
This is the "magic" that breaks up the words in ngrams. In theory this could be moved to mod.rs and exposed. Then it could be used to parse dictionaries the user provides instead of requiring them to have ngram lists.